central/graphql/resolvers/node_vulnerabilities.go

```diff
@@ -80,8 +82,366 @@ func (resolver *Resolver) NodeVulnCounter(ctx context.Context, args RawQuery) (*
+		return nil, err
+	}
+
+	q := search.EmptyQuery()
+	q.Pagination = &v1.QueryPagination{
+		Limit:  1,
+		Offset: 0,
+		SortOptions: []*v1.QuerySortOption{
+			{
+				Field:    search.NodeScanTime.String(),
+				Reversed: true,
+			},
+		},
+	}
+
+	nodes, err := nodeLoader.FromQuery(resolver.withNodeVulnerabilityScope(ctx), q)
+	if err != nil || len(nodes) == 0 {
+		return nil, err
+	} else if len(nodes) > 1 {
+		return nil, errors.New("multiple nodes matched for last scanned node vulnerability query")
+	}
+
+	return timestamp(nodes[0].GetScan().GetScanTime())
+}
+
+// Link to the node CVE
+func (resolver *nodeCVEResolver) Link(ctx context.Context) string {
+	return resolver.data.GetCveBaseInfo().GetLink()
+}
+
+// PublishedOn is date and time when this node CVE was first published in the cve feeds
+func (resolver *nodeCVEResolver) PublishedOn(ctx context.Context) (*graphql.Time, error) {
+	value := resolver.data.GetCveBaseInfo().GetPublishedOn()
+	return timestamp(value)
+}
+
+// ScoreVersion of the node CVE
+func (resolver *nodeCVEResolver) ScoreVersion(ctx context.Context) string {
+	value := resolver.data.GetCveBaseInfo().GetScoreVersion()
+	return value.String()
+}
+
+// Summary of the node CVE
+func (resolver *nodeCVEResolver) Summary(ctx context.Context) string {
+	return resolver.data.GetCveBaseInfo().GetSummary()
+}
+
+// SuppressActivation returns the snooze start timestamp of the node CVE
+func (resolver *nodeCVEResolver) SuppressActivation(ctx context.Context) (*graphql.Time, error) {
+	value := resolver.data.GetSnoozeStart()
+	return timestamp(value)
+}
+
+// SuppressExpiry returns the snooze expiration timestamp of the node CVE
+func (resolver *nodeCVEResolver) SuppressExpiry(ctx context.Context) (*graphql.Time, error) {
+	value := resolver.data.GetSnoozeExpiry()
+	return timestamp(value)
+}
+
+// Suppressed returns true if the node CVE is snoozed
+func (resolver *nodeCVEResolver) Suppressed(ctx context.Context) bool {
+	return resolver.data.GetSnoozed()
+}
+
+// UnusedVarSink represents a query sink
+func (resolver *nodeCVEResolver) UnusedVarSink(ctx context.Context, args RawQuery) *int32 {
+	return nil
+}
+
+// Vectors returns the CVSSV3 or CVSSV2 associated with the node CVE
+func (resolver *nodeCVEResolver) Vectors() *EmbeddedVulnerabilityVectorsResolver {
+	if val := resolver.data.GetCveBaseInfo().GetCvssV3(); val != nil {
+		return &EmbeddedVulnerabilityVectorsResolver{
+			resolver: &cVSSV3Resolver{resolver.ctx, resolver.root, val},
+		}
+	}
+	if val := resolver.data.GetCveBaseInfo().GetCvssV2(); val != nil {
+		return &EmbeddedVulnerabilityVectorsResolver{
+			resolver: &cVSSV2Resolver{resolver.ctx, resolver.root, val},
+		}
+	}
+	return nil
+}
+
+// VulnerabilityState returns the effective state of the node CVE (observed, deferred or marked as false positive).
+func (resolver *nodeCVEResolver) VulnerabilityState(ctx context.Context) string {
+	//TODO Should this be removed from nodeVulnerabilities graphQL ?
```

Comment:
@md2119

Reply:
IIRC supression is only for image vulns, I didn't think node vulns would have snooze functions either but they ended up there after data model transition

Reply:
Done